flink-core/src/test/java/org/apache/flink/api/common/typeutils/SerializerTestBase.java

```diff
@@ -409,9 +402,26 @@ else if (should instanceof Throwable) {
 			assertEquals(((Throwable)should).getMessage(), ((Throwable)is).getMessage());
 		}
 		else {
+			// if toString() was overridden we must check it
+			if (!should.toString().equals(should.getClass().getName() +
+										  "@" + Integer.toHexString(should.hashCode()))) {
```

Comment:
This can be more robustly done by looking up the 'toString()' method and seeing if the defining class was `Object`.